.github/workflows/maven.yml

```diff
@@ -0,0 +1,30 @@
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Java CI with Maven
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
```

Comment:
How about `ubuntu-latest` ? That should be more future proof

Reply:
Thank you for the feedback.

I had to reduce the number of points in that specific test so that it would pass with JDK 8; I *think* the test is invariant of the number of points, but I may be incorrect; hence this discussion is really useful.

With respect to ubuntu-latest; that's cool as well; I generally just enumerate the LTS versions of Ubuntu.

Reply:
Thank you!

Changing the number of points is fine, but actually in order for the test to use less memory I would recommend to lower the value coordsLimit. This is the maximum dimension of the volume that is generated here, so halving it should lead to ~1/8th memory usage.

Reply:
Thank you. However, resetting the nPoints  back to the original value of 1000 and setting coordsLimit to 48, results in that specific test failing locally.

Reply:
I see, probably because of the highest grid_width, which is 16, so the smallest volume becomes 3*3*3.

Could you please check that setting coordsLimit to 48 together with, e.g. 
> int nWidths = ResidueVolumeCache.GRID_WIDTHS.length-2; // to exclude the highest width

would work? Sorry I am not able to check this myself atm.

Reply:
Just tested that (nPoints = 1000, coordsLimit = 48); it also fails too.

Reply:
@mjw99 do you know how much memory you are allocating for the JVM?

Reply:
I think according to [this](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources) Actions hardware has 7GB of ram. However, I do not know exactly how the JVM is being invoked in that environment.

Reply:
That's odd. If I try configure the maven surefire plugin for this test with:
```xml
<plugin>
          <groupId>org.apache.maven.plugins</groupId>
          <artifactId>maven-surefire-plugin</artifactId>
          <configuration>
            <argLine>-Xmx2g</argLine>
          </configuration>
        </plugin>
```
Then the test passes with the original 1000 points. The test fails with 1000 points only when I change to `Xmx1g`

Reply:
Is that with JDK8?

Reply:
JDK 11

Reply:
IIRC; I was seeing the memory issues specifically with JDK 8

Reply:
Ok now tested with java 8 and for 1000 points it fails with Xmx2000m, but works with Xmx2500m